cql3/restrictions/statement_restrictions.cc

```diff
@@ -1357,6 +1477,23 @@ void statement_restrictions::prepare_indexed(const schema& idx_tbl_schema, bool
     }
     // If we're here, it means the index cannot be on a partition column: process_partition_key_restrictions()
     // avoids indexing when _partition_range_is_simple.  See _idx_tbl_ck_prefix blurb for its composition.
+
+    const column_definition* token_column = &idx_tbl_schema.clustering_column_at(0);
+
+    // In old indexes the token column was of type blob.
+    // This causes problems with sorting and must be handled separately.
+    _uses_v1_index = token_column->type != long_type;
+
+    if (has_token(_partition_key_restrictions->expression)) {
+        // When there is a token(p1, p2) >/</= ? restriction, it is not allowed to have restrictions on p1 or p2.
+        // This means that p1 and p2 can have arbitrary values (token is a hash, can have collisions).
```

Comment:
p1 and p2 can't have "arbitrary values", they need to have the expected hash function - maybe you meant to say there can be multiple matching partitions?

Reply:
No, it just means we can't know the values for partition columns, and we cannot generate individual restrictions for them.

Reply:
I changed this to `many different values`